RFC0032.md

```diff
@@ -0,0 +1,112 @@
+
+* PMIX\_ERR\_PARTIAL\_COMPLETION is returned if any channel fails to log as requested, but others succeed;
+* PMIX\_ERR\_OPERATION\_FAILED is returned if all fail; and
+* PMIX\_SUCCESS is returned if all succeed.
+
+This provides flexibility with minimal code complexity when operating in multiple environments that support differing output channels.
+
+Logging attributes can also utilize the "required" flag in the pmix\_info\_t structure to indicate that the data _must_ be logged via the specified channel. If given, failure to complete the operation on that channel will result in return of the PMIX\_ERR\_OPERATION\_FAILED error. Otherwise, use of a given channel is considered "optional" and errors are reported according to the above rules.
+
+Specifying a prioritized list of logging channels on each call to PMIx\_Log can impact the performance of the API itself as it requires the PMIx library to scan available channels to create an ordered list, and this might in turn require multiple passes over the available options. The PMIx reference library provides an MCA parameter to help reduce this impact. A user can control the default order of channel delivery by setting the "plog\_base\_order" MCA parameter to a comma-delimited, prioritized list of channel names based on the corresponding attribute by extracting the characters following "PMIX\_LOG\_", as follows:
+
+* PMIX\_LOG\_LOCAL\_SYSLOG - "local\_syslog"
+* PMIX\_LOG\_GLOBAL\_SYSLOG - "global\_syslog"
+* PMIX\_LOG\_EMAIL} - "email"
+
+and so on. Marking a given channel in the list as "required" can be done by adding ":req" to the channel name - e.g., plog\_base\_order = "local\_syslog:req,global\_syslog,email". Parsing of this MCA parameter is case-insensitive, and the parser will accept amy "required" flag that starts with "req" - e.g., "reqd" and "required". Similarly, the PMIX\_LOG\_ONCE attribute can be set by default using the "plog\_base\_log\_once" MCA parameter. Note that this is specific to the PMIx reference library and is _not_ part of the standard - users are advised to check their local implementation for similar support.
+
+Channels that are not recognized by the PMIx library will automatically be directed to the host RM for processing. This allows for RM-proprietary channel support without committing those channel names to the PMIx Standard.
+
+---
+
+**Advice to users:**  
+The available channel support on a system can be queried via the PMIx\_Query\_nb API should the application developer wish to tailor their code accordingly - this will always report the channels directly supported by the PMIx library. However, channels supported by the host RM will be included only if the RM itself supports such queries.
+
```

Comment:
How?

Reply:
I will add the attribute for that request